mlflow/pyfunc/__init__.py

```diff
@@ -847,7 +848,23 @@ def predict(*args):
         else:
             return result[result.columns[0]]
 
-    return pandas_udf(predict, result_type)
+    udf = pandas_udf(predict, result_type)
+
+    @functools.wraps(udf)
+    def udf_with_default_cols(*args):
+        if len(args) == 0:
+            model = SparkModelCache.get_or_load(archive_path)
```

Comment:
Rather than loading the full model, can we read the model yaml file from `local_model_path` (line 780)? 
I think it may be safer to avoid loading the model on the driver.

Reply:
I think you can load the Model while the tempdir exists (e.g. add the `model = Model.load(local_model_path)` to the with block on line 780) and use it later.

Reply:
in fact, it may be useful to attach the model object to the returned function object, e.g. `udf.metadata = model`